### PR TITLE
Feature: WebCodecs + WebGL video

### PR DESCRIPTION
**Pull request by gozmanyoni**
This PR is based on https://github.com/codewithpassion/foxglove-studio-h264-extension/tree/main

I re-used the worker from there to directly decode h264 frames and draw them on a canvas using WebGL - this replaces jmuxer, and offloads decoding to hardware + a separate thread.

### CODE REVIEW

File: examples/carplay-web-app/src/worker/Render.worker.ts [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+export type StatusType = 'render' | 'decode'
+type PartialRecord<K extends string, T> = Partial<Record<K, T>>
+export type StatusUpdate = PartialRecord<StatusType, string>
+
+// eslint-disable-next-line no-restricted-globals
+const scope = self as unknown as Worker
+
+type HostType = Window & typeof globalThis
+
+export class RenderWorker {
```
Comment (steelbrain) [inline examples/carplay-web-app/src/worker/Render.worker.ts:28]: There's quite a few usages of `.bind(this)` in this class, they can be eliminated by defining the class members in an alternative way, ie

```diff
-   private onVideoDecoderOutput(frame: VideoFrame) {
+   private onVideoDecoderOutput = (frame: VideoFrame) => {
```

**Review by steelbrain — COMMENTED** [review]
Thank you for working on this @gozmanyoni. This is incredible!

This is not measured through any scientific method but here's the rough comparison to jmuxer, across both macOS and Raspberry pi 4, rendering to canvas is noticably faster and responsive at the cost of a 5-10% increase in CPU usage on raspi but given the increase in performance and how smooth all the animations are, I'd say it's worth it.

I continue to be impressed by all the progress in this project :)

**Comment by gozmanyoni** [comment]
@rhysmorgan134 @steelbrain I also suspect with this we might not need the keyFrame interval

**Comment by rhysmorgan134** [comment]
super super exciting! Hows the performance difference?

**Review by rhysmorgan134 — APPROVED** [review]

**Review by steelbrain — APPROVED** [review]
Let's give this a go! Thank you @gozmanyoni

**Comment by gozmanyoni** [comment]
@rhysmorgan134 from what I have noticed:

* video stream starts up faster (now need to mux at all, so i assume we get a stream as soon as there is a keyframe)
* video overall is snappier
* interface is more responsive (main UI thread is a lot less busy now)

Did not measure in any scientific way though - the worker has an internal FPS indicator so we could print that and also calculate fps with jmuxer.

**Pull request merged**